zmq/auth/certs.py

```diff
@@ -102,6 +106,12 @@ def load_certificate(filename):
             line = line.strip()
             if line.startswith(b'#'):
                 continue
+            if load_metadata and line.startswith(b'metadata'):
```

Comment:
Is it valid for a file to define metadata after the keys? If so, this won't always work. Can you also add some tests?

Reply:
Thats indeed true. If the meta data is defined after the key, this won't work.

However, as far as I know, every certificate, which is created by zeromq has the same format:

```
metadata
    my_metadata1 = "172.17.64.145"
    my_metadata2 = "192.168.56.1"
    my_metadata3 = "130.75.27.242"
curve
    public-key = "2H0gQQr{^i6BUL:re4H{pb)b+LI:qtMMYvvl}1yT"
    secret-key = "YShO$Irv2}WM&jow1Kz.Oq$2!B6P4?f?/v^LYe<"
```

Am I wrong about this? If so, I will fix it. Do you want me to add tests anyways?

Reply:
That may be true in practice, but [the spec](https://rfc.zeromq.org/spec/4/) doesn't indicate an order is required. It might be appropriate to add a generic ZPL parser if we are starting to consume more than one section field, then call that `read_zpl(path)` in `load_cert`. What do you think?

Reply:
I like the idea of a generic ZPL parser. I will design one and introduce it either here, or in another PR.

What do you think, where we should place such a parser within this repo?

Reply:
Let's put it in utils as a private utility function, e.g. `utils.zpl.read_zpl(path) -> dict` then call it here.

Reply:
Sounds good. I will close this PR for now and create a new one, when I am done.

Reply:
There is actually already the python package [pyzpl](https://github.com/mbarkhau/pyzpl) on PyPi with under MIT license with that kind of functionality. We might either just use some of that code or merge that repository to pyzmq. What do you think?

Reply:
Hm, knowing that pyzpl exists and assuming it works, maybe the best thing to do is point people to that if they want more detailed loading of more fields instead of integrating it here?

Reply:
The pyzpl package seems to be well maintained and does what it is supposed to do. I tested it for my own needs.

However, I think this kind of functionality should be available in the "official" pyzmq package. But that is just me.

Reply:
That's a good point. For now, since pyzmq has no internal support for using information in fields other than the keys, let's point folks to pyzpl for reading zpl full files. We can revisit if there's a case to be made for support using metadata or other fields in the existing auth methods, etc.

Reply:
Alright. Then I will close this issue for now.